### PR TITLE
Add lint detector for missing smartling directives in String files

### PR DESCRIPTION
**Pull request by karlenDimla**
Task/Issue URL: https://app.asana.com/0/488551667048375/1203189982895630/f

### Description
- Lint check to ensure that all Smartling required directives are added in all String files.

### Steps to test this PR
Lint should pass.

### CODE REVIEW

**Comment by karlenDimla** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2485** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/2485" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/2485?utm_source=stack-comment).

File: lint-rules/src/main/java/com/duckduckgo/lint/MissingSmartlingRequiredDirectivesDetector.kt [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        document: Document
+    ) {
+        if (ignoreFile(context)) {
+            return
+        }
+
+        val contents = context.getContents() ?: return
+
+        REQUIRED_SMARTLING_DIRECTIVES.forEach {
+            if (!contents.contains(it)) {
```
Comment (marcosholgado) [inline lint-rules/src/main/java/com/duckduckgo/lint/MissingSmartlingRequiredDirectivesDetector.kt:53]: The rule is not really making sure the directives are created before resources, moving on directive anywhere in the file still passes the tests when it should not. Instead, you can iterate over the different nodes to make sure the directives exist before the resources node.

Comment (karlenDimla) [reply lint-rules/src/main/java/com/duckduckgo/lint/MissingSmartlingRequiredDirectivesDetector.kt:53]: Ahh! Good point. Completely missed that. Will update.

**Review by marcosholgado — APPROVED** [review]
Love it! Thanks @karlenDimla!

**Pull request merged**